DA/Internal/Desugar.hs

```diff
@@ -103,11 +107,35 @@ class HasFromAnyChoice t c r | t c -> r where
 class HasInterfaceTypeRep i where
   interfaceTypeRep : i -> TypeRep
 
-class HasInterfaceTypeRep i => Implements t i where
-  toInterface : t -> i
+class HasToInterface t i where
+  _toInterface : t -> i
+
+toInterface : forall i t. HasToInterface t i => t -> i
```

Comment:
why do we need a separate top-level method if it’s identical to the typeclass method?

Reply:
https://github.com/digital-asset/daml/pull/11964#discussion_r763229493